### DISCUSSION TITLE
How to open a NDArray for read and write?

### DISCUSSION THREAD

**Discussion by Karol-G** [Blosc/python-blosc2#178] [General]
Hey,

I am having difficulties understanding the documentation on how to create a peristent NDArray on disk that is populated slice-wise with data over time. The same goes vice-versa with reading from the NDArray on the disk. 

With Zarr I would approach it like this:

Writing:
```python
import zarr
import numpy

array_zarr = zarr.open("tmp.zarr", shape=(10, ), dtype=np.float32, chunks=(2, ), mode='w')
array_zarr[:5] = [1, 2, 1, 2, 1]
array_zarr[5:] = [3, 4, 3, 4, 3]
```

Reading:
```python
import zarr
import numpy

array_zarr = zarr.open("tmp.zarr", mode='r')
array = np.zeros((10, ))
array[:5] = array_zarr[:5]
array[5:] = array_zarr[5:]
```

How do I do the same with blosc2?

Best regards,
Karol

**Comment by FrancescAlted** [comment]
Sure.  Following the tutorial should help you grasping the basics for persistence and slicing: https://www.blosc.org/python-blosc2/getting_started/tutorials/02.ndarray-basics.html

For creating an empty array (the equivalent of open in zarr), you can use [blosc2.empty()](https://www.blosc.org/python-blosc2/reference/autofiles/ndarray/blosc2.empty.html) with a `urlpath` as keyword argument.

BTW, the documentation can be improved for sure.  If you think that you can contribute to the cause, we would gladly accept PRs.

**Comment by Karol-G** [comment]
Thank you for the quick answer!
